recirq/qcqmc/hamiltonian.py

```diff
@@ -0,0 +1,319 @@
+from pathlib import Path
```

Comment:
This file is missing the license.  I noticed this is part 3.  Did we miss licenses in parts 1 and 2?

Reply:
1 was ok, 2 was the external (vendored) dependency so I just added a license to the directory (but not to the individual files). 4 was missing the license which I added here.